aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -43,10 +43,23 @@ struct StashTLSOnEntryGuard {
   c10::impl::LocalDispatchKeySet saved_;
 };
 
-void pythonFallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+void pythonCompositeFallback(const c10::OperatorHandle& op, c10::DispatchKeySet ks, torch::jit::Stack* stack) {
+  c10::impl::ExcludeDispatchKeyGuard guard_(
+      c10::DispatchKeySet({c10::DispatchKey::Python, c10::DispatchKey::PythonComposite}));
+  op.callBoxed(stack);
+}
+
+void pythonFallback(const c10::OperatorHandle& op, c10::DispatchKeySet ks, torch::jit::Stack* stack) {
   TORCH_INTERNAL_ASSERT(tls_on_entry.has_value());
   // c10::impl::ForceDispatchKeyGuard dispatcher_guard(tls_on_entry.value());
   // StashTLSOnEntryGuard stash_guard;
+
+  // TODO: use guard semantics instead, see https://github.com/pytorch/pytorch/pull/74728
+  if (at::impl::TorchDispatchModeTLS::exchange_skip_next(false)) {
+    op.redispatchBoxed(ks.remove(c10::DispatchKey::Python), stack);
```

Comment:
Normally when you call `redispatch(ks, ...)`, the intent is to redispatch to "whatever the next-highest-priority kernel is".

In this case though, we basically know that we want to hit the `PythonComposite` kernel. Do you think it's worth making this `redispatchBoxed(DispatchKeySet(DispatchKey::PythonComposite), ...)` just to make intent clearer?

Reply:
Fair enough

Reply:
Isn't that ok only if ks doesn't contain any key below PythonComposite ? What would happen if the PythonComposite kernel redispatch as well?

Reply:
In this case it would definitely be safe because I don't use the passed in dispatch key set in PythonComposite, and you would revert back to consulting the TLS. But I'm also happy with keeping the logic here as is.